tests/ui/associated-types/associated-types-eq-2.stderr

```diff
@@ -1,9 +1,177 @@
+error[E0658]: associated const equality is incomplete
+  --> $DIR/associated-types-eq-2.rs:52:10
+   |
+LL | impl Tr3<N = 42, T2 = Qux, T3 = usize> for Bar {
+   |          ^^^^^^
+   |
+   = note: see issue #92827 <https://github.com/rust-lang/rust/issues/92827> for more information
+   = help: add `#![feature(associated_const_equality)]` to the crate attributes to enable
+   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
+
+error[E0229]: associated type bindings are not allowed here
+  --> $DIR/associated-types-eq-2.rs:17:31
+   |
+LL | fn baz<I: Foo>(_x: &<I as Foo<A=Bar>>::A) {}
+   |                               ^^^^^ associated type not allowed here
+   |
+   = note: `Foo` is not a generic type
+help: try removing this type binding
+   |
+LL | fn baz<I: Foo>(_x: &<I as Foo<A=Bar>>::A) {}
+   |                              ~~~~~~~
+
+error[E0107]: trait takes 1 generic argument but 0 generic arguments were supplied
+  --> $DIR/associated-types-eq-2.rs:24:6
+   |
+LL | impl Tr1<T1 = String> for Bar {
+   |      ^^^ expected 1 generic argument
+   |
+note: trait defined here, with 1 generic parameter: `T1`
+  --> $DIR/associated-types-eq-2.rs:21:7
+   |
+LL | trait Tr1<T1> {
+   |       ^^^ --
+help: add missing generic argument
+   |
+LL | impl Tr1<T1, T1 = String> for Bar {
+   |          +++
```

Comment:
Ideally we wouldn't emit this note which gets emitted by `lower_generic_args` in module `generics` but that might be tricky to implement.

We could think about implementing the “use the type|const directlyy” suggestion inside `lower_generic_args` instead of `prohibit_assoc_item_binding` :thinking:

Reply:
Yeah, I didn't like this and the new suggestions popping up together. I'll see if we can do something about them including implementing the new ones in `lower_generic_args`.

Reply:
Not done this yet. May tackle in a separate PR.